src/NuGet.Clients/NuGet.PackageManagement.UI/Options/AddMappingDialog.xaml.cs

```diff
@@ -0,0 +1,111 @@
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Windows;
+using System.Windows.Input;
+using Microsoft.ServiceHub.Framework;
+using NuGet.PackageManagement.UI;
+using NuGet.VisualStudio;
+using NuGet.VisualStudio.Common;
+using NuGet.VisualStudio.Internal.Contracts;
+using Task = System.Threading.Tasks.Task;
+
+
+namespace NuGet.Options
+{
+    public partial class AddMappingDialog : VsDialogWindow
+    {
+        public ICommand HideButtonCommand { get; set; }
+
+        public ICommand AddButtonCommand { get; set; }
+
+        public ItemsChangeObservableCollection<PackageSourceContextInfoChecked> SourcesCollection { get; private set; }
+
+        private IReadOnlyList<PackageSourceContextInfo> _originalPackageSources;
+
+#pragma warning disable ISB001 // Dispose of proxies, disposed in disposing event or in ClearSettings
+        private INuGetSourcesService _nugetSourcesService;
+#pragma warning restore ISB001 // Dispose of proxies, disposed in disposing event or in ClearSettings
+
+        private PackageSourceMappingOptionsControl _parent;
+
+        public AddMappingDialog(PackageSourceMappingOptionsControl parent)
+        {
+            _parent = parent;
+            HideButtonCommand = new ButtonCommand(ExecuteHideButtonCommand, CanExecuteHideButtonCommand);
+            AddButtonCommand = new ButtonCommand(ExecuteAddButtonCommand, CanExecuteAddButtonCommand);
+            SourcesCollection = new ItemsChangeObservableCollection<PackageSourceContextInfoChecked>();
+            DataContext = this;
+            InitializeComponent();
+            CancellationToken cancellationToken = new CancellationToken(false);
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () => await InitializeOnActivatedAsync(cancellationToken));
+        }
+
+        internal async Task InitializeOnActivatedAsync(CancellationToken cancellationToken)
+        {
+            IServiceBrokerProvider serviceBrokerProvider = await ServiceLocator.GetComponentModelServiceAsync<IServiceBrokerProvider>();
+            IServiceBroker serviceBroker = await serviceBrokerProvider.GetAsync();
+#pragma warning disable ISB001 // Dispose of proxies, disposed in disposing event or in ClearSettings
+            _nugetSourcesService = await serviceBroker.GetProxyAsync<INuGetSourcesService>(
```

Comment:
If you are only using `_nugetSourcesService` member variable inside this method, I think you can remove the member variable to avoid keeping a reference.

Reply:
Good point, and I'm also now Disposing it: https://github.com/NuGet/NuGet.Client/commit/847ac9cd0cf71b2fdca01c1b1936c25ea8fcc695